### PR TITLE
Document Zold::Upgrades #343

### PR DESCRIPTION
**Pull request by botanicus**
This is the `Zold::Upgrades` documentation requested by @ledestin in #343.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#353](https://github.com/zold-io/zold/issues/353) is now in scope, role is `REV`

File: lib/zold/upgrades.rb [hunk 1]
```diff
@@ -2,6 +2,22 @@
 
 module Zold
   # Class to manage data upgrades (when zold itself upgrades).
+  #
+  # This class will write the version file to the zoldata directory.
```
Comment (ledestin) [inline lib/zold/upgrades.rb]: @botanicus what happened to the draft? This text doesn't explain the case of missing version file.

Comment (botanicus) [reply lib/zold/upgrades.rb]: @ledestin added.

**Comment by botanicus** [comment]
@ledestin added. Is it good now?

File: lib/zold/upgrades.rb [hunk 2]
```diff
@@ -2,6 +2,26 @@
...
+  # of the data and a newline. It is named `version`, no extension.
+  #
+  # If the data is up-to-date, the version of the data is equal to
+  # Zold::VERSION.
+  #
+  # If the version is lower than Zold::VERSION, upgrade scripts from
+  # `upgrades/` have to run. They are named `<version>.rb`, so for
+  # instance `upgrades/0.0.1.rb` etc.
+  #
+  # Only the scripts from the data version up need to run.
```
Comment (ledestin) [inline lib/zold/upgrades.rb]: @botanicus greater than the data version? Or >=?

**Comment by ledestin** [comment]
@botanicus thanks, but one more question

File: lib/zold/upgrades.rb (hunk 2 above)
Comment (botanicus) [reply lib/zold/upgrades.rb]: If we're on 0.1 and want to migrate to 0.2, we should run `0.2.rb`, `0.1.rb` run already when we upgraded to 0.1, I'd say. So I think this is correct.

Comment (ledestin) [reply lib/zold/upgrades.rb]: @botanicus I have a hard time understanding it, and one of my mates too. Could you please change it in the docs?

**Comment by 0crat** [comment]
@ledestin[/z](https://www.0crat.com/u/ledestin) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by botanicus** [comment]
@ledestin sorry, I'm traveling and didn't have time. Let me rephrase it.

**Comment by botanicus** [comment]
@ledestin I just changed that.

File: lib/zold/upgrades.rb [hunk 3]
```diff
@@ -10,11 +10,18 @@ module Zold
...
-  # instance `upgrades/0.0.1.rb` etc.
+  # By comparing Zold::VERSION with the data version we determine
+  # which upgrade scripts will be executed.
   #
-  # Only the scripts from the data version up need to run.
+  # If the data version is the same as Zold::VERSION, the data is
+  # up to date.
+  #
+  # If the data version is lower than Zold::VERSION, we will look into
+  # the `upgrades/` directory and any scripts from the data version
```
Comment (ledestin) [inline lib/zold/upgrades.rb]: @botanicus please don't use `data version up`, it's ambiguous at least to me. You can also ask @yegor256 to resolve our conflict.
